doc/man3/SSL_get_session.pod

```diff
@@ -26,7 +26,30 @@ count of the B<SSL_SESSION> is incremented by one.
 =head1 NOTES
 
 The ssl session contains all information required to re-establish the
-connection without a new handshake.
+connection without a full handshake for SSL versions <= TLSv1.2. In TLSv1.3 the
+same is true, but sessions are established after the main handshake has occurred.
+The server will send the session information to the client at a time of its
+choosing which may be some while after the initial connection is established (or
+not at all). Calling these functions on the client side in TLSv1.3 before the
+session has been established will still return an SSL_SESSION object but it
+cannot be used for resuming the session. See L<SSL_SESSION_is_resumable(3)> for
+information on how to determine whether an SSL_SESSION object can be used for
+resumption or not.
+
+Additionally, in TLSv1.3, a server can send multiple session messages for a
+single connection. In that case the above functions will only return information
+on the last session that was received.
```

Comment:
> If you come back later, the callback may be called again in the meantime. That seems weird. It also means the session returned by SSL_get_session changes over time.

Right - but that can happen now with <= TLSv1.2 in a renegotiation - and we can't change that behaviour.

> By set, you mean if you SSL_set_session while configuring a SSL and immediately afterwards calling SSL_get_session? I'm not proposing breaking that one.

Ah - ok. Thanks for the clarification. Yes, that's what I meant.